src/include/duckdb.h

```diff
@@ -2483,6 +2483,23 @@ Note that the object must be destroyed with `duckdb_appender_destroy`.
 DUCKDB_API duckdb_state duckdb_appender_create(duckdb_connection connection, const char *schema, const char *table,
                                                duckdb_appender *out_appender);
 
+/*!
+Creates an array of DUCKDB_TYPEs that represent the types of the columns of the table that the appender appends to.
+
+The resulting array should be freed with `duckdb_appender_destroy_column_types`.
+
+* appender: The appender to get the column types from.
+* out_types: The resulting array of types.
+*/
+DUCKDB_API duckdb_state duckdb_appender_get_column_types(duckdb_appender appender, enum DUCKDB_TYPE **out_types);
```

Comment:
I think the following signatures would be better/more versatile:

```cpp
idx_t duckdb_appender_column_count(duckdb_appender appender);
duckdb_logical_type duckdb_appender_column_type(duckdb_appender appender, idx_t index);
```

DUCKDB_TYPE is not sufficient for nested types - and by just returning a single type at a time we don't need a special destructor function.